src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,113 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
+                    }
                 }
             });
     }
 }
 
+void AutoSchedule::releaseActualdevice(std::string currentDeviceName) {
+    _loadContext[ACTUALDEVICE].isEnabled = false;
+    _loadContext[ACTUALDEVICE].isLoadSuccess = false;
+    _loadContext[ACTUALDEVICE].isAlready = false;
+    _loadContext[ACTUALDEVICE].future.wait();
+    // clean up ACTUALDEVICE infer requests
+    size_t destroynum = 0;
+    std::pair<int, WorkerInferRequest*> worker;
+    while (_idleWorkerRequests[currentDeviceName].try_pop(worker)) {
+        destroynum++;
+    }
+    if (destroynum == _workerRequests[currentDeviceName].size()) {
+        std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+        _workerRequests[currentDeviceName].clear();
+        _loadContext[ACTUALDEVICE].executableNetwork._ptr.reset();
+        _loadContext[ACTUALDEVICE].executableNetwork._so.reset();
+    }
+    LOG_INFO_TAG("ACTUALDEVICE released!!");
+}
+
+bool AutoSchedule::selectOtherDevice(std::string currentDeviceName) {
+    {
+        std::lock_guard<std::mutex> lock(_autoSContext->_fallbackMutex);
+        if (_autoSContext->_modelPath.empty())
+            _loadContext[FALLBACKDEVICE].networkPrecision = GetNetworkPrecision(_autoSContext->_network);
+        _loadContext[FALLBACKDEVICE].metaDevices = _autoSContext->_devicePriorities;
+        if (currentDeviceName == "CPU_HELP") {
+            currentDeviceName = "CPU";
+        }
+        const auto CurrentDeviceIter = std::find_if(_autoSContext->_devicePriorities.begin(), _autoSContext->_devicePriorities.end(),
+                                            [=](const DeviceInformation& d) -> bool {
+                                            return d.deviceName.find(currentDeviceName) != std::string::npos;});
+        if (CurrentDeviceIter != _autoSContext->_devicePriorities.end()) {
+            if (_autoSContext->_devicePriorities.size() == 1) {
+                LOG_INFO_TAG("No other devices in _devicePriorities");
+                return false;
+            }
+            _autoSContext->_devicePriorities.erase(CurrentDeviceIter);
+        } else {
+            LOG_INFO_TAG("Alreay selected fallback device");
+            return _loadContext[FALLBACKDEVICE].isReloadSuccess ? true : false;
+        }
+        _loadContext[FALLBACKDEVICE].isLoadSuccess = false;
+        _loadContext[FALLBACKDEVICE].workName = "";
+        _loadContext[FALLBACKDEVICE].isReloadSuccess = false;
+        _loadContext[FALLBACKDEVICE].deviceInfo =
+            _autoSContext->_plugin->SelectDevice(_autoSContext->_devicePriorities,
+                                                    _loadContext[FALLBACKDEVICE].networkPrecision,
+                                                    _autoSContext->_modelPriority);
+        _executor->run(_loadContext[FALLBACKDEVICE].task);
+        std::unique_lock<std::mutex> lck(_mutex);
+        _cond.wait(lck);
+        if (_loadContext[FALLBACKDEVICE].isAlready) {
```

Comment:
btw, with the _fallbackMutex, is it possible that other threads may even run to reach this line?

Reply:
Other threads can not reach this line with _fallbackMutex.

Reply:
so I suppose the call_once is not needed?

Reply:
also, for optimization, I want to see if this thread can share common code with cpu_help release, with the target release device name as parameter